From 1e4711af5b98717563f45bbce350b592c77ca646 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 30 Mar 2020 15:42:02 +0200
Subject: [PATCH] userwork: fix signal worker sends to manager requesting more
 workers

That's what you get for changing these signals around, after checking
everything works, not before.

Bad, Lennart, bad!

Fixes: #15085
(cherry picked from commit e3028514aeb7feee67b4c4747a3af52d09a3e9fd)
---
 src/userdb/userwork.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/userdb/userwork.c b/src/userdb/userwork.c
index df11b5b984e..3bc5ecc1d03 100644
--- a/src/userdb/userwork.c
+++ b/src/userdb/userwork.c
@@ -763,7 +763,7 @@ static int run(int argc, char *argv[]) {
                                 if (parent <= 1)
                                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Parent already died?");
 
-                                if (kill(parent, SIGUSR1) < 0)
+                                if (kill(parent, SIGUSR2) < 0)
                                         return log_error_errno(errno, "Failed to kill our own parent.");
                         }
                 }
